internal/cmd/server.go

```diff
@@ -175,15 +173,19 @@ func setupRunners(cfg *config.Server) error {
 		return err
 	}
 
-	// Start the global rateLimit runner if it has been enabled through the config
+	// Start the global rateLimit if it has been enabled through the config
 	if cfg.EnvoyGateway.RateLimit != nil {
-		// Start the Global RateLimit Runner
-		// It subscribes to the xds Resources and translates it to Envoy Ratelimit Service
-		// infrastructure and configuration.
+		// Create the global ratelimit infra statically.
+		mgr := infraRunner.Manager()
+		if err := mgr.CreateOrUpdateRateLimitInfra(ctx); err != nil {
```

Comment:
prefer to  move to `runner`

Reply:
+1 , prefer if this lives inside the infra-manager runner
Also you can block on `<-ctx.Done()` in https://github.com/envoyproxy/gateway/blob/56edabaac958d34254028b8fb9f0f68ed3c1ac9d/internal/infrastructure/runner/runner.go#L51
and delete the RL service if it was created before

Reply:
sounds really great